utils/python/CIME/test_scheduler.py

```diff
@@ -80,7 +80,7 @@ def __init__(self, test_names, test_data=None,
             self._test_root = self._test_root.replace("$PROJECT", self._project)
 
         self._test_root = os.path.abspath(self._test_root)
-        self._test_id   = test_id if test_id is not None else CIME.utils.get_timestamp()
+        self._test_id   = test_id
```

Comment:
I'm not sure about this change. The default value for test_id is None. Don't we need to handle the case where it's None by assigning it to something? We don't need to add the random component here.